bacon/cubequery.py

```diff
@@ -148,14 +166,25 @@ def remove_filter(self, name, value=None, operator=None):
 			rv._filters = [f for f in rv._filters if f != (name, operator, value)]
 		return rv
 
-	def invert_filter(self, name, value, operator):
-		"""Return a new `CubeQuery` with filter inverted."""
+	def swap_filter(self, name, value, operator, new_operator):
+		"""Return a new `CubeQuery` with filter operator changed."""
 		rv = self.copy()
 		original = (name, operator, value)
-		inverted = (name, invert_op(operator), value)
-		rv._filters = [inverted if f == original else f for f in rv._filters]
+		replacement = (name, new_operator, value)
+		rv._filters = [replacement if f == original else f for f in rv._filters]
 		return rv
 
+	def invert_filter(self, name, value, operator):
+		"""Return a new `CubeQuery` with filter inverted."""
+		return self.swap_filter(name, value, operator, invert_op(operator))
+
+	def related_filters(self, name, value, operator):
+		"""Return a dictionary of new `CubeQuery`s with other ops in place of this filter."""
+		return {
+			other_op: self.swap_filter(name, value, operator, other_op)
+			for other_op in related_ops(operator)
```

Comment:
The current operator will be included in this. We could return the set without it in (related_ops)

Reply:
Yeah, I was contemplating removing that, plus the inverse (which has its own thing), but actually I think it "feels" better to have the same drop-down regardless of the op-set used.